library/std/src/io/stdio.rs

```diff
@@ -820,6 +828,9 @@ impl Write for StdoutLock<'_> {
     fn write_all_vectored(&mut self, bufs: &mut [IoSlice<'_>]) -> io::Result<()> {
         self.inner.borrow_mut().write_all_vectored(bufs)
     }
+    fn write_fmt(&mut self, fmt: fmt::Arguments<'_>) -> io::Result<()> {
+        self.inner.borrow_mut().write_fmt(fmt)
```

Comment:
We don't do this because the formatting implementations might try to use `Stdout` themselves, which would lead to panicking.

Reply:
Thanks. I had overlooked that this could call user code, and was only thinking about locking concerns on the stdlib side. I dropped that commit now.